### ISSUE TITLE
support the MERG table

### ISSUE THREAD

**Issue by matthiasclasen** [harfbuzz/harfbuzz#4405]
In GTK, we recently ran into the issue that the MERG table is meant to solve:

![image](https://github.com/harfbuzz/harfbuzz/assets/4834845/9c86fcc9-f8b1-48e5-a9ef-6605c9ee0ac7)

It would be nice to have some solution for this, even if not many fonts have it.

Sadly, it seems that web browsers (except for webkit) just ignore this in the name of performance.

@jfkthame

**Comment by khaledhosny** [comment]
I don't think many people know this table exists, let alone ignore it.

**Comment by behdad** [comment]
Yeah it's a tough one. We can implement the table and add API to take a shaped buffer and return a bool. Matthias had this idea that if the table is missing we can fallback to a script-based solution.

**Comment by behdad** [comment]
> Yeah it's a tough one. We can implement the table and add API to take a shaped buffer and return a bool. Matthias had this idea that if the table is missing we can fallback to a script-based solution.

The fallback would also work as the (previously requested) "can this script be letterspaced".

**Comment by matthiasclasen** [comment]
The gtk issue mentions a few cases where merg might not help: https://gitlab.gnome.org/GNOME/gtk/-/issues/6075

such as O, followed by U0336:

![image](https://github.com/harfbuzz/harfbuzz/assets/4834845/b3c723b5-c05b-4b47-b65f-a3a8fd0d26e5)

**Comment by khaledhosny** [comment]
IIUC, the webkit solution is more general and doesn't need any extra font-side data:
https://faultlore.com/blah/text-hates-you/#text-overlaps

**Comment by behdad** [comment]
Always using an off-screen for translucent text might be best.

**Comment by matthiasclasen** [comment]
Unfortunately, current GNOME design makes basically all text translucent :(

**Comment by DemiMarie** [comment]
@matthiasclasen can this be fixed in GNOME?

**Comment by KrasnayaPloshchad** [comment]
I’ve seen the similar issue affecting Fontworks’ rendering in LibreOffice.
https://bugs.documentfoundation.org/show_bug.cgi?id=104594
With this support it’s probably helpful to make the fix for LibreOffice.
